gufe/components/proteincomponent.py

```diff
@@ -1,86 +1,671 @@
+                    atom.GetAtomicNum(),
+                    name,
+                    atom.GetFormalCharge(),
+                    atom.GetIsAromatic(),
+                    "",  # Stereocent in Smallcomponent
+                    atom.GetPropsAsDict(),
+                    mi_dict,
+                )
+            )
+
+        bonds = [
+            (
+                bond.GetBeginAtomIdx(),
+                bond.GetEndAtomIdx(),
+                bond.GetBondType(),
+                bond.GetIsAromatic(),
+                bond.GetStereo() or "",
+                bond.GetPropsAsDict(),
+            )
+            for bond in self._rdkit.GetBonds()
+        ]
+
+        conformers = [
+            serialize_numpy(conf.GetPositions())  # .m_as(unit.angstrom)
+            for conf in self._rdkit.GetConformers()
+        ]
+
+        # Additional Information for the mol:
+        molecule_props = {}
+        for prop_key, prop_value in self._rdkit.GetPropsAsDict(
+            includePrivate=True
+        ).items():
+            if prop_key == "sequence":
+                residue_sequence = prop_value.split()
+                molecule_props["sequence"] = residue_sequence
+            elif isinstance(prop_value, str) and prop_value.startswith("{"):
+                val = json.loads(prop_value.replace("'", '"'))
+                molecule_props[prop_key] = val
+            elif isinstance(prop_value, str) and prop_value.startswith("["):
+                val = ast.literal_eval(prop_value)
+                molecule_props[prop_key] = val
+            else:
+                molecule_props[prop_key] = prop_value
+
+        # Result
+        d = {
+            "atoms": atoms,
+            "bonds": bonds,
+            "name": self.name,
+            "conformers": conformers,
+            "molecules": molecule_props,
+        }
+
+        return d
+
+    # NOT implemented:
+    def to_openeye(self) -> OEMol:
+        """OEChem representation of this molecule"""
+        raise NotImplementedError
+
+    def to_openff(self):
+        raise NotImplementedError
+
+    def to_gmx(self, out_gmx_top: str, out_gmx_gro: str):
+        raise NotImplementedError
+
+    def to_amber(self, out_path: str):
+        raise NotImplementedError
+
+    @classmethod
+    def from_openeye(cls, oemol: OEMol, name: str = ""):
+        raise NotImplementedError
+
+    @classmethod
+    def from_openff(cls, openff: OFFMolecule, name: str = ""):
+        raise NotImplementedError
 
     @classmethod
-    def _from_dict(cls, d: dict):
-        raise NotImplementedError()
+    def from_gmx(cls, gmx_top: str, gmx_gro: str):
+        raise NotImplementedError
 
-    @property
-    def total_charge(self):
-        return None
+    @classmethod
+    def from_amber(cls, name=""):
+        raise NotImplementedError
 
-    def _defaults(self):
-        return super()._defaults()
+    @classmethod
+    def from_pdbxfile(cls, pdbxfile: str, name=""):
+        raise NotImplemented
```

Comment:
did this not work? It looked like all the code for it should have been implemented.

Reply:
In theory it should be there, but somehow, there is a problem (maybe with our testfile?)

Reply:
![image](https://user-images.githubusercontent.com/12428005/191003456-f0d38e99-2d98-4e58-b096-27ec81e24a25.png)

something is weird with the connectivity.... as it is 0 for this atom (which it definitly should not be and is correctly assigned in Topology, somehow it is overwritten somewhere?/unlinked?)

Reply:
hm.... ok I got it, the atom names were not recognized by openmm.... fixed.